scripts/annotate_sam_with_bed_features.py

```diff
@@ -0,0 +1,328 @@
+#!/usr/bin/env python
+
+# pylint: disable=line-too-long
+
+"""Annotate SAM alignments with their intersecting feature(s).
+
+Add a custom tag ("YW") to each alignment in the SAM file if an intersecting
+feature is found in the BED file. The BED file must be the result of a
+bedtools intersect call where `-a` is a GFF3 file and `-b` a BAM file. If
+either the BED or the SAM file is empty, only the SAM file header is returned.
+
+Each matching feature is used to build a tag with the following format:
+
+    FEATURE_ID|5'-shift|3'-shift|CIGAR|MD|READ_SEQ.
+
+Where:
+    - FEATURE_ID: Extracted from the specified attribute in the BED file
+    (default: "name").
+    - 5'-shift: Difference between the (possibly adjusted) feature start and
+    the alignment start.
+    - 3'-shift: Difference between the alignment end and the (possibly
+    adjusted) feature end.
+    - CIGAR and MD: The alignment's CIGAR string and MD tag respectively.
+    - READ_SEQ: The read sequence from the alignment.
+
+Optional adjustments:
+--extension: Adjust the feature's start and end coordinates by the given value
+    (start is increased and end decreased).
+--shift: Requires both the 5' and 3' shift values to be within +/- this value
+    to include the feature tag in the alignment.
+
+If an alignment has multiple intersecting features, the tag values are
+concatenated using a semicolon as the separator. If no valid intersecting
+feature is found, the alignment is skipped.
```

Comment:
Is this last sentence accurate? It seems to me that the alignment is only skipped if there are no intersecting features _or_ if none of the intersecting features pass the shift filter.